autokeras/hypermodel/block.py

```diff
@@ -0,0 +1,595 @@
+                                                          default='global_ave')
+
+        if reduction_type == 'flatten':
+            output_node = Flatten().build(hp, output_node)
+        elif reduction_type == 'global_max':
+            output_node = tf.math.reduce_max(output_node, axis=-2)
+        elif reduction_type == 'global_ave':
+            output_node = tf.math.reduce_mean(output_node, axis=-2)
+        elif reduction_type == 'global_min':
+            output_node = tf.math.reduce_min(output_node, axis=-2)
+
+        return output_node
+
+
+class EmbeddingBlock(HyperBlock):
+    """Word embedding block for sequences.
+
+    The input should be tokenized sequences with the same length, where each element
+    of a sequence should be the index of the word.
+
+    Attributes:
+        pretrained: Boolean. Use pretrained word embedding. Defaults to tunable.
+        is_embedding_trainable: Boolean. Defaults to tunable.
+        embedding_dim: Int. Defaults to None.
+    """
+
+    def __init__(self,
+                 pretrained=None,
+                 is_embedding_trainable=None,
+                 embedding_dim=None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.pretrained = pretrained
+        self.is_embedding_trainable = is_embedding_trainable
+        self.embedding_dim = embedding_dim
+
+    def build(self, hp, inputs=None):
+        input_node = nest.flatten(inputs)[0]
+        pretrained = self.pretrained or hp.Choice('pretrained',
+                                                  [True, False],
+                                                  default=False)
+        is_embedding_trainable = self.is_embedding_trainable or hp.Choice(
+            'is_embedding_trainable',
+            [True, False],
+            default=False)
+        embedding_dim = self.embedding_dim or hp.Choice('embedding_dim',
+                                                        [32, 64, 128, 256, 512],
+                                                        default=128)
+        if pretrained:
+            # TODO: load from pretrained weights
+            layer = tf.keras.layers.Embedding(
+                input_dim=input_node.shape[1],
+                output_dim=embedding_dim,
+                input_length=const.Constant.VOCABULARY_SIZE,
+                trainable=is_embedding_trainable)
+            # weights=[embedding_matrix])
+        else:
+            layer = tf.keras.layers.Embedding(
+                input_dim=input_node.shape[1],
+                output_dim=embedding_dim,
+                input_length=const.Constant.VOCABULARY_SIZE,
+                trainable=is_embedding_trainable)
+        return layer(input_node)
+
+
+class TextBlock(RNNBlock):
+    pass
+
+
+class StructuredBlock(HyperBlock):
+
+    def build(self, hp, inputs=None):
+        pass
+
+
+class TimeSeriesBlock(HyperBlock):
+
+    def build(self, hp, inputs=None):
+        pass
+
+
+class GeneralBlock(HyperBlock):
```

Comment:
The user may not specify the type of the input since their input may not fall into any of the categories we defined. They would want this general block if they don't know anything about what architecture to use.

Reply:
How would that be supported in practice?